demo: Example app with calendar assistant using tool calling

pweglik opened the pull request:
## Description

<!-- Provide a concise and descriptive summary of the changes implemented in this PR. -->

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improves or adds clarity to existing documentation)

### Tested on

- [x] iOS
- [x] Android

### Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly
- [ ] My changes generate no new warnings

### Additional notes

<!-- Include any additional information, assumptions, or context that reviewers might need to understand this PR. -->

NorbertKlockiewicz commented on examples/llm-tool-calling/src/screens/tools.ts:
```diff
@@ -0,0 +1,212 @@
+    (event) => `${event.title}, from: ${event.startDate}, to: ${event.endDate}`
+  );
+  return eventsStringRepresentation.join('\n');
+};
```
Missing enter

NorbertKlockiewicz commented on examples/llm-tool-calling/metro.config.js:
```diff
@@ -0,0 +1,22 @@
+  };
+
+  config.resolver.assetExts.push('pte');
+  config.resolver.assetExts.push('bin');
```
I mean this line: `config.resolver.assetExts.push('bin');`

NorbertKlockiewicz commented:
I think we should add some response message if tool was called, e.g "Tool execution finished successfully" or smth like this as for now the message generated by llm just disappears and it's a bit confusing form me

NorbertKlockiewicz commented:
Also maybe some predefined prompt for user would be great as I barely know what should I tell the model.

pweglik commented:
> I think we should add some response message if tool was called, e.g "Tool execution finished successfully" or smth like this as for now the message generated by llm just disappears and it's a bit confusing form me
We have param for it called: `displayToolCalls`. I rhink I will set it to true. Normally showing tool call is undesired but in demo app it is confusing, you're right. `i set it to true, when calling `useLLM` from `ChatScreen`
>Also maybe some predefined prompt for user would be great as I barely know what should I tell the model.
Great recommendation, I'll add something

pweglik commented:
<img width="433" alt="Screenshot 2025-04-28 at 09 52 10" src="https://github.com/user-attachments/assets/03c27c07-733e-466d-9131-aff169702cc1" />
Added text and removed unnecessary code.
